### PR TITLE
egress loss: average stats from intermediate receiver reports

### PR DESCRIPTION
**Pull request by davibe**
Before this commit we were calculating stats from latest RR received. Turns out browsers send RRs much more frequently especially in case of network problems (10+ per second). Downsampling to the latest every second looses a lot of information considering that each RR reports the fraction lost from the previous one.

With this patch we are averaging the values weighted against the amount of packet seen by the receiver, which makes much more sense in the case of a periodic update.

### CODE REVIEW

File: str0m/src/media/sender.rs [hunk 1]
```diff
@@ -148,15 +149,46 @@ impl SenderSource {
         let rtt = calculate_rtt_ms(ntp_time, r.last_sr_delay, r.last_sr_time);
         self.rtt = rtt;
 
-        self.loss = Some(r.fraction_lost as f32 / u8::MAX as f32);
+        let ext_seq = {
+            let ext = (r.max_seq >> 16) as u64;
+            let num = (r.max_seq & 0xffff) as u16;
+            extend_seq(Some(ext), num)
```
Comment (algesten) [inline str0m/src/media/sender.rs:155]: The `Some(ext)` part here must be saved on ReceiverSource and reused for every incoming RR.

**Review by algesten — CHANGES_REQUESTED** [review]

**Review by algesten — APPROVED** [review]
